micronaut/src/main/java/software/amazonaws/example/product/PrimingResource.java

```diff
@@ -0,0 +1,34 @@
+package software.amazonaws.example.product;
+
+import io.micronaut.crac.OrderedResource;
+import jakarta.inject.Singleton;
+import org.crac.Context;
+import org.crac.Resource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazonaws.example.product.dao.DynamoProductDao;
+
+@Singleton
+public class PrimingResource implements OrderedResource {
+  private static final Logger log = LoggerFactory.getLogger(PrimingResource.class);
+  private final DynamoProductDao dynamoProductDao;
+
+  public PrimingResource(DynamoProductDao dynamoProductDao) {
+    this.dynamoProductDao = dynamoProductDao;
```

Comment:
Spring Boot pattern sets the `Core.getGlobalContext().register(this);`, is this something we need here too?

Reply:
`micronaut-crac` does that under the hood.